### ISSUE TITLE
user_profile is undefined on message from USLACKBOT (TypeError: Cannot read property 'name' of undefined)

### ISSUE THREAD

**Issue by andybalaam** [matrix-hacks/matrix-puppet-slack#40]
When I receive certain messages from USLACKBOT I see an error `TypeError: Cannot read property 'name' of undefined`

Stack trace:

```
::ffff:10.9.64.188 - - [08/Jun/2018:15:59:47 +0000] "PUT /transactions/637?access_token=<REDACTED> HTTP/1.1" 200 2 "-" "Synapse/0.31.0"

{ type: 'message',
  user: 'USLACKBOT',
  text: 'Reminder: <!subteam^S801BRMME|@teamname> Tokeniser needs tokens! Nom nom nom. <https://redacted.example.com/>',
  bot_id: 'B01',
  channel: 'C7YFX4JV6',
  team: 'T0251NS0H',
  event_ts: '1528473626.000006',
  ts: '1528473626.000006' }
/home/andrebal/code/matrix-puppet-slack/node_modules/olm/olm.js:19
"undefined"!==typeof module&&(module.exports=a);process.on("uncaughtException",function(a){if(!(a instanceof T))throw a;});a.inspect=function(){return"[Emscripten Module object]"}}else if(ma)a.print||(a.print=print),"undefined"!=typeof printErr&&(a.printErr=printErr),a.read="undefined"!=typeof read?read:function(){throw"no read() available (jsc?)";},a.readBinary=function(a){if("function"===typeof readbuffer)return new Uint8Array(readbuffer(a));a=read(a,"binary");z("object"===typeof a);return a},"undefined"!=
                                                                                                                ^

TypeError: Cannot read property 'name' of undefined
    at App.getPayload (/home/andrebal/code/matrix-puppet-slack/app.js:107:43)
    at App.createAndSendPayload (/home/andrebal/code/matrix-puppet-slack/app.js:225:24)
    at Client.client.on (/home/andrebal/code/matrix-puppet-slack/app.js:80:16)
    at emitOne (events.js:116:13)
    at Client.emit (events.js:211:7)
    at RTMClient.rtm.on (/home/andrebal/code/matrix-puppet-slack/client.js:63:16)
    at RTMClient.emit (/home/andrebal/code/matrix-puppet-slack/node_modules/eventemitter3/index.js:116:35)
    at RTMClient.handleWsMessage (/home/andrebal/code/matrix-puppet-slack/node_modules/@slack/client/lib/clients/rtm/client.js:423:8)
    at WebSocket.wrapper (/home/andrebal/code/matrix-puppet-slack/node_modules/lodash/lodash.js:4949:19)
    at emitTwo (events.js:126:13)
    at WebSocket.emit (events.js:214:7)
    at Receiver.ontext (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/WebSocket.js:841:10)
    at /home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:536:18
    at Receiver.applyExtensions (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:371:5)
    at /home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:508:14
    at Receiver.flush (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:347:3)
    at Receiver.finish (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:541:12)
    at Receiver.expectHandler (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:499:31)
    at Receiver.add (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/Receiver.js:103:24)
    at TLSSocket.realHandler (/home/andrebal/code/matrix-puppet-slack/node_modules/ws/lib/WebSocket.js:825:20)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
```

**Comment by d3m3vilurr** [comment]
it was fixed in #45 ;)

**Comment by thomas-profitt** [comment]
Thanks for the activity. I got the ping, tested this, and it is indeed now working as expected 💓
